### ISSUE TITLE
ostris/minimax_h3_training_adapter

### ISSUE THREAD

**Issue by Jamzsam** [tdrussell/diffusion-pipe#534]
ostris has just released a minimax_h3_training_adapter although it is an Alpha version. (If this helps at all.)

https://huggingface.co/ostris/minimax_h3_training_adapter/tree/main

**Comment by paju1986** [comment]
It can be used with diffusion-pipe, i tried it and it works. but i couldnt get very good likeness on my character lora yet. Im tying different settings.

**Comment by Jamzsam** [comment]
[paju1986
Yeah I'm having the same problem. Using only 29 images for training my character is 80% there but starts over fitting before likeness is 100%. Video turns to Mush. I'm trying a run at the moment with (image_shift = 1.25) to see if it improves.

**Comment by PlutonB** [comment]
> It can be used with diffusion-pipe, i tried it and it works. but i couldnt get very good likeness on my character lora yet. Im tying different settings.

Can you give some detail on how to use the adapter with diffusion-pipe?

**Comment by paju1986** [comment]
# Will train MiniMax H3 LoRAs on 24GB VRAM, and has reasonable defaults.

# FILL IN THESE PATHS
# change this
output_dir = '/mnt/HDD3T/datasets/valeriedyson/1024/'
# and this
dataset = '/mnt/AI/diffusion-pipe/dataset.toml'

 #You can have separate eval datasets. Give them a name for Tensorboard metrics.
 eval_datasets = [
     {name = 'datasetVal', config = '/mnt/AI/diffusion-pipe/datasetEval.toml'},
 ]

# training settings
epochs = 30
micro_batch_size_per_gpu = 1
image_micro_batch_size_per_gpu = 1  # 8 images in a batch uses about as much VRAM as a single 5-second video
pipeline_stages = 1
gradient_accumulation_steps = 4
gradient_clipping = 1
warmup_steps = 50

# eval settings
eval_every_n_epochs = 1
#eval_every_n_steps = 100
eval_before_first_step = true
eval_micro_batch_size_per_gpu = 1
eval_gradient_accumulation_steps = 1

# misc settings
save_every_n_epochs = 1  # use either this or the steps one below
#save_every_n_steps = 200
checkpoint_every_n_epochs = 1
#checkpoint_every_n_minutes = 120
activation_checkpointing = 'unsloth'
partition_method = 'parameters'
save_dtype = 'bfloat16'
caching_batch_size = 1
steps_per_print = 1
video_clip_mode = 'single_beginning'
#blocks_to_swap = 24  # 48 is max. At 512 res, might be able to lower this a bit to speed things up.
#compile = true

# FILL IN ALL THESE PATHS TOO
[model]
type = 'minimax_h3'
diffusion_model = '/mnt/AI/ComfyUI/models/diffusion_models/minimax/minimax_h3_fl2va_pruned_int8_convrot.safetensors'
vae = '/mnt/AI/ComfyUI/models/vae/minimax_h3_video_vae_fp16.safetensors'
audio_vae = '/mnt/AI/ComfyUI/models/vae/minimax_h3_audio_vae_fp32.safetensors'
text_encoders = [
    {path = '/mnt/AI/ComfyUI/models/text_encoders/qwen3vl_32b_minimax_h3_int8_convrot.safetensors', type = 'minimax'}
]
dtype = 'bfloat16'
timestep_sample_method = 'uniform'  # or logit_normal, IDK which is better
shift = 8  # experiment with this
image_shift = 1  # 1, or slightly higher than 1, seems good for images
merge_adapters = ['/mnt/AI/diffusion-pipe/minimax_h3_training_adapter_v1.safetensors']  # merges adapters into weights before training starts

[adapter]
type = 'lora'
rank = 32
dtype = 'bfloat16'

#[optimizer]
#type = 'adamw_optimi'
#lr = 1e-4
#betas = [0.9, 0.99]
#weight_decay = 0.01
#eps = 1e-8

[optimizer]
type = 'automagic'
weight_decay = 0.01

**Comment by paju1986** [comment]
Im not getting good likeness still. any ideas what is wrong? the same dataset was used for krea2 with excelent results

**Comment by Jamzsam** [comment]
> I’m not getting good likeness still. any ideas what is wrong? the same dataset was used for krea2 with excellent results

So raising image_shift = 1 to image_shift = 1.25 definitely helped identity. I would say 90% there.  My best result is Epoch 80 at 1120 steps using 29 images. I’ve noticed that because i trained at 768 resolution generating at 0.6mp or 0.7mp resolution yields better likeness and identity. Anything higher destroys the likeness and identity. I am thinking if it could train at multiple resolutions in a training run such as resolutions = [512],[768],[1024] it maybe more flexible but I’m not sure how to setup the config for multiple resolutions. I might try raising image_shift = 1.25 to image_shift = 1.50 in my next run to see if it improves.
I hope this helps in any way.

**Comment by paju1986** [comment]
Im been trying to train a character lora with 190 images all weekend, i tried several learning rates, cfg method, adapter model method. but all no luck, horrible likeness something must be wrong, same dataset gived excelent results with krea2...

**Comment by Jamzsam** [comment]
> Im been trying to train a character lora with 190 images all weekend, i tried several learning rates, cfg method, adapter model method. but all no luck, horrible likeness something must be wrong, same dataset gived excelent results with krea2...

Yep same here. Multi resolutions came out bad. This model is definitely different than previous video models. I'm busy trying  image_shift = 2, lr = 2e-4, and resolutions = [1024]. I'm trying random settings at this point. Hopefully some one out there can crack this puzzle.

**Comment by rkfg** [comment]
I used `image_shift = 1.5`, `lr = 1e-4`, `resultions = [768]`, `rank = 16`, masked training (masks created in OneTrainer, then renamed to match the dataset file names). The dataset consists of 97 images. I trained for 30 epochs and that seems to be too much, anatomy and physics break. The 20th epoch looks good though. The similarity is pretty close in T2V but my goal is using it in R2V with some references, so that the lora improves small details and overall subject knowledge that's missing in those few images I provide. I agree that it's not as good at learning people as the older models. But the model itself is so powerful I'm sure it all will improve soon.

**Comment by paju1986** [comment]
I tried almost all possible parameters but there is now way i get likeness at least with my krea2 dataset, that it worked perfectly for krea2. This models is way worse at learning. :(

**Comment by rkfg** [comment]
Did you recaption it to follow the T2V prompt format?

**Comment by paju1986** [comment]
no idea what is that? i had ollama captioning the images

**Comment by rkfg** [comment]
Bruh moment ﻿😂﻿ H3 is extremely sensitive to the prompt format and contents, that's what makes it so good. The prompt (caption) should look like this:
```
integrated_multimodal_description: [Shot 1] bla-bla-bla, the image described as a video preferably, we pretend it's a video, not a still frame

overall_soundscape: N/A

non_diegetic_music: N/A
```
No audio so `N/A` is valid. If you provide a verbatim prompt like you'd do for any other model, it causes overfit because the model doesn't expect it and corrects the error where there's none to learn the new prompt format. Instead, it should localize the loss around your subject and that requires the prompt to be "expected". Also, use masking, it prevents training the background to focus only on the subject, I like how mask editing is implemented in OneTrainer but it's possible to do in any image editor (will take more time probably). Just rename `image-masklabel.png` to `image.png` to match the original image name and specify the masks directory in the dataset config. Read the config format and README.

**Comment by paju1986** [comment]
Interesting, i just tried to train it like any other model i didnt know it was so picky, i will try with your formula hopefully i get better results, thanks!

**Comment by Dream-Making-Git** [comment]
https://huggingface.co/ostris/minimax_h3_training_adapter/blob/main/minimax_h3_training_adapter_v2.safetensors

looks like he posted an update as well
